### PR TITLE
Dev v4.0.0

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Description

- min supported version bump from ios 10 to ios 11
- min xcode version supported bumped to xcode 14.1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -22,9 +22,9 @@ def lib_main
 end
 
 def lib_dev
-    pod 'AEPCore', :git => 'https://github.com/adobe/aepsdk-core-ios.git', :branch => 'dev-v3.7.3'
```
Comment (rymorale) [inline Podfile:25]: the specified branch could be removed right? the core dependency could be 
```
pod 'AEPCore', '~> 4.0'
```

File: .github/workflows/update_versions.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ on:
     # Inputs the workflow accepts.
     inputs:
       version:
-        description: 'New version to use for the Messaging extension.  Example: 3.0.2'
+        description: 'New version to use for the Messaging extension.  Example: 4.0.2'
```
Comment (swarna04) [inline .github/workflows/update_versions.yml:12]: nit: why 4.0.2? `4.0.0`

File: AEPMessaging.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -2193,13 +2192,12 @@
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				INFOPLIST_FILE = AEPMessaging/Sources/Info.plist;
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
```
Comment (swarna04) [inline AEPMessaging.xcodeproj/project.pbxproj:2196]: Shouldn't it have also added `IPHONEOS_DEPLOYMENT_TARGET = 11.0;`

File: Script/update-versions.sh [hunk 4]
```diff
@@ -31,8 +31,8 @@ help()
    echo "Usage: $0 -n EXTENSION_NAME -v NEW_VERSION -d \"PODSPEC_DEPENDENCY_1, PODSPEC_DEPENDENCY_2\""
    echo ""
    echo -e "    -n\t- Name of the extension getting a version update. \n\t  Example: Edge, Analytics\n"
-   echo -e "    -v\t- New version to use for the extension. \n\t  Example: 3.0.2\n"
-   echo -e "    -d (optional)\t- Dependency(ies) that require updating in the extension's podspec and Package.swift file. \n\t  Example: -d \"AEPCore 3.7.3\" (update the dependency on AEPCore to version 3.7.3 or newer)\n"
+   echo -e "    -v\t- New version to use for the extension. \n\t  Example: 4.0.2\n"
+   echo -e "    -d (optional)\t- Dependency(ies) that require updating in the extension's podspec and Package.swift file. \n\t  Example: -d \"AEPCore 4.0.2\" (update the dependency on AEPCore to version 4.0.2 or newer)\n"
```
Comment (swarna04) [inline Script/update-versions.sh:35]: 4.0.2 => 4.0.0

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: Podfile (hunk 1 above)
Comment (sbenedicadb) [reply Podfile:25]: this is just for the `lib_dev` group.  the podfile isn't actually using this group at all right now, only using `lib_main`

File: AEPMessaging.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sbenedicadb) [reply AEPMessaging.xcodeproj/project.pbxproj:2196]: 11.0 is the project setting.  10.0 was a target override.  the remove of this line causes the target to use the value set in the project (11.0)

**Review by swarna04 — APPROVED** [review]

**Pull request merged**